python/build-wheel-windows.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)"
+WORKSPACE_DIR="${ROOT_DIR}/.."
+
+PY_VERSIONS=(3.6 3.7 3.8)
+PY_SCRIPT_SUBDIR=Scripts  # 'bin' for UNIX, 'Scripts' for Windows
+
+get_python_version() {
+  python -s -c "import sys; sys.stdout.write('%s.%s' % sys.version_info[:2])"
+}
+
+is_python_version() {
+  local expected result=0
+  expected="$1"
+  case "$(get_python_version).0." in
+    "${expected}".*) ;;
+    *) result=1;;
+  esac
+  case "$(pip --version | tr -d "\r")" in
+    *" (python ${expected})") ;;
+    *) result=1;;
+  esac
+  return "${result}"
+}
+
+build_wheel_windows() {
+  local ray_uninstall_status=0
+  pip uninstall -y ray || ray_uninstall_status=1
+
+  local pyversion pyversions=()
+  for pyversion in "${PY_VERSIONS[@]}"; do
+    if [ "${pyversion}" = "${PYTHON-}" ]; then continue; fi  # we'll build ${PYTHON} last
+    pyversions+=("${pyversion}")
+  done
+  pyversions+=("${PYTHON-}")  # build this last so any subsequent steps use the right version
+
+  local local_dir="python/dist"
+  for pyversion in "${pyversions[@]}"; do
+    if [ -z "${pyversion}" ]; then continue; fi
+    "${WORKSPACE_DIR}"/ci/travis/bazel-preclean.sh
+    git clean -q -f -f -x -d -e "${local_dir}" -e python/ray/dashboard/client
+    git checkout -q -f -- .
+
+    # Start a subshell to prevent PATH and cd from affecting our shell environment
+    (
+      if ! is_python_version "${pyversion}"; then
+        local pydirs=("${RUNNER_TOOL_CACHE}/Python/${pyversion}".*/x64)
+        local pydir="${pydirs[-1]}"
+        pydir="$(cygpath -u "${pydir}")"  # Translate Windows path
+        test -d "${pydir}"
+        export PATH="${pydir}:${pydir}/${PY_SCRIPT_SUBDIR}:${PATH}"
+      fi
+      if ! is_python_version "${pyversion}"; then
+        echo "Expected pip for Python ${pyversion} but found Python $(get_python_version) with $(pip --version); exiting..." 1>&2
+        exit 1
+      fi
+
+      unset PYTHON2_BIN_PATH PYTHON3_BIN_PATH  # make sure these aren't set by some chance
+      install_ray
+      cd "${WORKSPACE_DIR}"/python
+      python setup.py --quiet bdist_wheel
+      pip uninstall -y ray
+    )
+  done
+
+  "${WORKSPACE_DIR}"/ci/travis/bazel-preclean.sh
+  if [ 0 -eq "${ray_uninstall_status}" ]; then  # If Ray was previously installed, restore it
+    install_ray
+  fi
```

Comment:
User can install un their own. Do we need this in our CI?

Reply:
This isn't for users, it's or our CI. Whether we build the wheels and then do testing is in an arbitrary order (currently we build wheels first, then test, but we can easily swap it around later) so this is needed to ensure any installation remains after wheels are built.